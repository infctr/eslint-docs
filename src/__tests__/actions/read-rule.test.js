jest.mock('../../spinner')

const spinner = require('../../spinner')

const chalk = require('chalk')
chalk.enabled = true

const readRule = require('../../actions/read-rule')

const description = 'Require that the file be empty'
const name = 'require-empty'
const docs =
  `
# ${description} (${name})

This rule ensures great productivity by requiring that
all files you write are completely empty.
  `.trim() + '\n'

const friendlyDocPath = `docs/rules/${chalk.bold(name)}`

const runReadRule = (
  rule = {
    meta: {
      docs: { description },
    },
  },
  docFile = docs
) =>
  readRule(
    {
      rule,
      docs: docFile,
      friendlyDocPath,
    },
    name
  )

describe('readRule()', () => {
  it('reads the description', () => {
    const ruleModule = {
      meta: {
        docs: {},
      },
    }
    const getDescription = jest.fn(() => description)
    Object.defineProperty(ruleModule.meta.docs, 'description', {
      get: getDescription,
    })

    runReadRule(ruleModule)

    expect(getDescription).toHaveBeenCalled()
  })

  it('does not modify the docs', () => {
    expect(runReadRule().newDocs).toBe(docs)
  })

  it('updates the docs if needed', () => {
    expect(runReadRule(undefined, docs.replace('# ', '# ***')).newDocs).toBe(
      docs
    )
    expect(spinner.info).toHaveBeenCalledWith(
      `Updating the docs for ${chalk.bold(name)}:`
    )

    const mockLog = console.log.mock
    expect(mockLog.calls[mockLog.calls.length - 1]).toMatchSnapshot()
  })
})
