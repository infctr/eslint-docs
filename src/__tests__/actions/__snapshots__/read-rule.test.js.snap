// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readRule() updates the docs if needed 1`] = `
Array [
  "  [31m--- docs/rules/[1mrequire-empty[22m[39m
  [32m+++ generated[39m
  [2m[34m@@ -1,4 +1,4 @@[39m[22m
  [31m-# ***Require that the file be empty (require-empty)[39m
  [32m+# Require that the file be empty (require-empty)[39m
   
   This rule ensures great productivity by requiring that
   all files you write are completely empty.
  ",
]
`;
